balancer/backends.go

```diff
@@ -157,5 +145,14 @@ func (b *backends) updateBackendScores() error {
 		}
 		succeeded = append(succeeded, addr)
 	}
-	return b.Update(toStrset(succeeded))
+	return b.Update(succeeded)
+}
+
+// --------------------------------------------------------------------
+
+type strset []string
+
+func (s strset) Contains(v string) bool {
+	pos := sort.SearchStrings(s, v)
```

Comment:
just inline these two lines, no need for strset or even an extra func, just `if pos := ... ; pos < len(s) && s[pos] == v {...}`